neqo-http3/src/connection.rs

```diff
@@ -130,14 +130,14 @@ impl<T: Http3Transaction> Http3Connection<T> {
     fn create_qpack_streams(&mut self, conn: &mut Connection) -> Res<()> {
         qdebug!([self], "create_qpack_streams.");
         self.qpack_encoder
-            .add_send_stream(conn.stream_create(StreamType::UniDi)?);
+            .add_send_stream(conn.stream_create(StreamType::UniDi)?.into());
```

Comment:
Having qpack know about StreamId isn't ideal, but I think that it probably should in the end.  Or, we could make these qpack public functions generic on `<T: Into<u64>>`.

Reply:
@martinthomson Thanks for the comments!
In many case I just used `as_u64` to make the code compile till I get back to these places: I was not sure whether I needed to touch collections or not. 


I did too many changes at once and it's kinda difficult to make sense of them now. I was thinking about closing this PR and starting from scratch: maybe trying to localize changes within a single package first and temporary fixing other places with `as_u64` where stream_id is expected to be u64.

Reply:
There was also one specific place I didn't know how to deal with.
If I change [these fields](https://github.com/mozilla/neqo/blob/master/neqo-qpack/src/decoder_instructions.rs#L19-L20) to `StreamId`, but leave `InsertCountIncrement { increment: u64 }` as is, how do I go about [this match](https://github.com/mozilla/neqo/blob/master/neqo-qpack/src/decoder_instructions.rs#L111) without adding arm for InsertCountIncrement separately?

Reply:
Eww, you are right, that's ugly.  It might be a case of this being too hard and we can factor this differently, at a small cost in ugliness:

```rust
                        qtrace!([self], "varint read {}", val);
                        let set_none = |dec: &mut Self| {
                            dec.state = DecoderInstructionReaderState::ReadInstruction;
                            Ok(Some(mem::replace(
                                &mut self.instruction,
                                DecoderInstruction::NoInstruction,
                            )))
                        };
                        match &mut self.instruction {
                            DecoderInstruction::InsertCountIncrement { increment: v } => {
                                *v = val;
                                break set_none(&mut self);
                            }
                            DecoderInstruction::HeaderAck { stream_id: v }
                            | DecoderInstruction::StreamCancellation { stream_id: v } => {
                                *v = StreamId::from(val);
                                break set_none(&mut self);
                            }
                            _ => unreachable!(...);
                        }
```

Or you could avoid this weird code and use the simple version instead:

```rust
                        qtrace!([self], "varint read {}", val);
                        let ret = match self.instruction {
                            DecoderInstruction::InsertCountIncrement { .. } =>
                                DecoderInstruction::InsertCountIncrement { increment: val },
                            DecoderInstruction::HeaderAck { stream_id: v } =>
                                DecoderInstruction::HeaderAck { stream_id: StreamId::from(val) },
                            DecoderInstruction::StreamCancellation { stream_id: v } =>
                                DecoderInstruction::StreamCancellation { stream_id: StreamId::from(val) },
                            _ => unreachable!(...);
                        };
                       self.instruction = DecoderInstruction::NoInstruction;
                       ret
```

Reply:
@martinthomson thanks a lot!